### ISSUE TITLE
Fix READMEs, make it easier to read code/spec

### ISSUE THREAD

**Issue by mwhicks1** [cedar-policy/cedar-spec#71]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/cedar-policy/cedar/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/cedar-policy/cedar/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Bug Category

Other

### Describe the bug

The README explaining this repository refers to directories that don't exist. Rename/update accordingly.

In addition, it would help to link references to these directories, and to likewise link references in `cedar-dafny/README.md` to appropriate files/lines of the code. Doing this will encourage users who might be a little unsure about looking at the code to click the link and take a look.

Finally, add a README in the `cedar-policy-generators` directory, for parity with the other subdirectories.

### Expected behavior

Docs should be better.

### Reproduction steps

None.

### Code Snippet

```rust
// Put your code below this line.

```

### Log output

```
// Put your output below this line

```

**Comment by khieta** [comment]
@shaobo-he-aws resolved by #74?

**Comment by shaobo-he-aws** [comment]
> @shaobo-he-aws resolved by #74?

Yes. Closed this issue.
